crypto/params.c

```diff
@@ -1488,11 +1488,11 @@ int OSSL_PARAM_get_octet_ptr(const OSSL_PARAM *p, const void **val,
 static int set_ptr_internal(OSSL_PARAM *p, const void *val,
                             unsigned int type, size_t len)
 {
-    p->return_size = len;
     if (p->data_type != type) {
         err_bad_type;
         return 0;
     }
+    p->return_size = len;
```

Comment:
Setting this early was deliberate.  These calls try to provide size info on failure.
This change should be reverted.

Reply:
This is wrong when the type is incorrect and can lead to crashes. Already discussed in detail with Richard, this is fine.

Reply:
Here's where I agree with @vdukhovni.  If the data type of the OSSL_PARAM isn't what this function expects, how can any return length be meaningful?
Do note that the return length has already been set to zero by the wrapping caller, if you're worried about that mechanism.